lib/common/xml.c

```diff
@@ -3009,6 +3009,21 @@ pcmk__xe_set_props(xmlNodePtr node, ...)
     va_end(pairs);
 }
 
+void
+pcmk__xe_foreach_child(xmlNode *xml, const char *child_element_name,
+                       void (*handler)(xmlNode *xml, void *userdata),
+                       void *userdata)
+{
+    CRM_ASSERT(xml != NULL && handler != NULL);
```

Comment:
I think xml==NULL should be ok (no-op). That would let us do something like

    pcmk__xe_foreach_child(first_named_child(cib, SOME_ELEMENT), SOME_SUBELEMENT, handler, data));

if SOME_ELEMENT is optional.